misc/openapi-ballerina/modules/openapi-validator/src/main/java/org/ballerinalang/openapi/validator/BTypeToJsonValidatorUtil.java

```diff
@@ -0,0 +1,785 @@
+
+                    } else if (entry.getValue() instanceof ObjectSchema) {
+                        //  Handle the nested record type
+                        validateNestedRecord(validationErrors, recordType, field, entry);
+                    } else {
+                        //  Handle array type mismatching.
+                        if (field.getValue().getType().getKind().typeName().equals("[]")) {
+                            BArrayType bArrayType = null;
+                            Schema entrySchema = entry.getValue();
+                            ArraySchema arraySchema = new ArraySchema();
+                            if ((field.getValue().type instanceof BArrayType) && (entrySchema instanceof ArraySchema)) {
+                                bArrayType = (BArrayType) field.getValue().type;
+                                arraySchema = (ArraySchema) entrySchema;
+                            }
+                            if (bArrayType != null) {
+                                BArrayType traversNestedArray = bArrayType;
+                                ArraySchema traversSchemaNestedArray = arraySchema;
+                                //  Handle nested array type
+                                if ((bArrayType.eType instanceof BArrayType) &&
+                                        (arraySchema.getItems() instanceof ArraySchema)) {
+                                    Schema traversSchemaNestedArraySchemaType = arraySchema.getItems();
+                                    traversNestedArray = (BArrayType) bArrayType.eType;
+
+                                    if (traversSchemaNestedArraySchemaType instanceof ArraySchema) {
+                                        traversSchemaNestedArray = (ArraySchema) traversSchemaNestedArraySchemaType;
+                                    }
+                                    while ((traversNestedArray.eType instanceof BArrayType) &&
+                                            (traversSchemaNestedArray.getItems() instanceof ArraySchema)) {
+                                        Schema<?> traversSchemaNestedArraySchema = traversSchemaNestedArray.getItems();
+                                        BType traversNestedArrayBtype = traversNestedArray.eType;
+
+                                        if ((traversSchemaNestedArraySchema instanceof ArraySchema) &&
+                                                (traversNestedArrayBtype instanceof BArrayType)) {
+                                            traversSchemaNestedArray = (ArraySchema) traversSchemaNestedArraySchema;
+                                            traversNestedArray = (BArrayType) traversNestedArrayBtype;
+                                        }
+                                    }
+                                }
+                                //  Handle record type in item array
+                                if ((traversNestedArray.eType instanceof BRecordType) &&
```

Comment:
There is a small typo. `traversNestedArray` has to be `traverseNestedArray`. `traverse` is misspelled everywhere